Incorrect ValidateDocumentationComments warning for rethrows function

`Sources/ArgumentParser/Utilities/Mutex.swift:53:5: warning: [ValidateDocumentationComments] remove the 'Throws:' sections of 'withLock'; it does not throw any errors`

```swift
  /// Calls the given closure after acquiring the lock and then releases
  /// ownership.
  ///
  /// - Warning: Recursive calls to `withLock` within the closure parameter has
  ///   behavior that is platform dependent. Some platforms may choose to panic
  ///   the process, deadlock, or leave this behavior unspecified. This will
  ///   never reacquire the lock however.
  ///
  /// - Parameter body: A closure with a parameter of `Value` that has exclusive
  ///   access to the value being stored within this mutex. This closure is
  ///   considered the critical section as it will only be executed once the
  ///   calling thread has acquired the lock.
  ///
  /// - Throws: Re-throws any error thrown by `body`.
  ///
  /// - Returns: The return value, if any, of the `body` closure parameter.
  func withLock<U>(
    _ body: (inout T) throws -> U
  ) rethrows -> U {
    self.lock.lock()
    defer { self.lock.unlock() }
    return try body(&self.value)
  }
```

--- a/Sources/SwiftFormat/Rules/ValidateDocumentationComments.swift
+++ b/Sources/SwiftFormat/Rules/ValidateDocumentationComments.swift
@@ -131,12 +131,13 @@ public final class ValidateDocumentationComments: SyntaxLintRule {
     throwsDescription: Paragraph?,
     node: DeclSyntax
   ) {
-    // If a function is marked as `rethrows`, it doesn't have any errors of its
-    // own that should be documented. So only require documentation for
-    // functions marked `throws`.
+    // Documentation is required for functions marked as `throws`.
+    // For functions marked as `rethrows`, documentation is not enforced
+    // since they don’t introduce new errors of their own.
+    // However, it can still be included if needed.
     let needsThrowsDesc = throwsOrRethrowsKeyword?.tokenKind == .keyword(.throws)
 
-    if !needsThrowsDesc && throwsDescription != nil {
+    if throwsOrRethrowsKeyword == nil && throwsDescription != nil {
       diagnose(
         .removeThrowsComment(funcName: name),
         on: throwsOrRethrowsKeyword ?? node.firstToken(viewMode: .sourceAccurate)

--- a/Tests/SwiftFormatTests/Rules/ValidateDocumentationCommentsTests.swift
+++ b/Tests/SwiftFormatTests/Rules/ValidateDocumentationCommentsTests.swift
@@ -95,7 +95,6 @@ final class ValidateDocumentationCommentsTests: LintOrFormatRuleTestCase {
       findings: [
         FindingSpec("1️⃣", message: "remove the 'Throws:' sections of 'doesNotThrow'; it does not throw any errors"),
         FindingSpec("2️⃣", message: "add a 'Throws:' section to document the errors thrown by 'doesThrow'"),
-        FindingSpec("3️⃣", message: "remove the 'Throws:' sections of 'doesRethrow'; it does not throw any errors"),
       ]
     )
   }

